### DISCUSSION TITLE
Not able to run on a local setup

### DISCUSSION THREAD

**Discussion by usaaama2010** [zitadel/zitadel#4883] [Q&A] [answered]
I am trying to setup Zitadel on my local system but every time i run the below command it gives me this error:

```
 ERROR: executor failed running [/bin/sh -c go install github.com/rakyll/statik     && go test -race -v -coverprofile=profile.cov $(go list ./... | grep -v /operator/)]: exit code: 1
------
 > [go-test 5/5] RUN go install github.com/rakyll/statik     && go test -race -v -coverprofile=profile.cov $(go list ./... | grep -v /operator/):
------
executor failed running [/bin/sh -c go install github.com/rakyll/statik     && go test -race -v -coverprofile=profile.cov $(go list ./... | grep -v /operator/)]: exit code: 1
```

I am using M1 Macbook and following this guide for local setup https://github.com/zitadel/zitadel/blob/main/CONTRIBUTING.md

Any help on this would be really appreciated, thanks :)

**Comment by fforootd** [comment, marked as answer]
Hi There

What docker version are you using? Please make sure you set the `DOCKER_BUILDKIT=1` because this sometimes can have strange effects if not set.

Besides that, if you only want to run ZITADEL I would advise using our released docker image. E.g with [docker compose.](https://zitadel.com/docs/guides/deploy/compose)

**Reply by usaaama2010** [reply]
Hi @fforootd

I'm using this command `goreleaser build --id dev --snapshot --single-target --rm-dist --output .artifacts/zitadel/zitadel` and my docker version is 20.10.21. Actually, I've made some modifications in code and want to build the docker image with update code, although i was able to build the docker image but when i run it then it gives me this error `exec /zitadel: exec format error`

**Reply by fforootd** [reply]
Got it.

Can you share the content of your .artifacts, folder?

**Reply by usaaama2010** [reply]
Here it is:
```
├── .artifacts
│   ├── zitadel
│   │   ├── .gitignore
│   │   ├── .gitkeep

2 files, 1 directory
```

**Reply by fforootd** [reply]
Ah, I got an idea. I think the problem could be the unit test being run on CockroachDB.

https://github.com/zitadel/zitadel/blob/main/build/zitadel/Dockerfile#L85

@livio-a @adlerhurst since you also have m1 chips, does that work for you, building locally?

**Reply by usaaama2010** [reply]
I have made some modifications in the goreleaser configs and updated the CRDB version to the latest and now i am able to build the image successfully but it is still giving this error: `exec /zitadel: exec format error`

**Reply by fforootd** [reply]
That sounds to me like a problem around the cpu architecture. 

Even though goreleaser with `--single-target` should use your local GOOS and GORACH settings.

**Reply by usaaama2010** [reply]
Yeah i am looking into it but in the meanwhile, i am also trying to run the application directly but when it runs it just gives me this error when i open http://localhost:8080 

Error: `The requested redirect_uri is missing in the client configuration. If you have any questions, you may contact the administrator of the application.`

**Reply by usaaama2010** [reply]
@fforootd - I solved the issue by modifying the goreleaser builds configs, if I just specify `goarch=amd64 goos=linux` then it works just fine, not sure if this is the right fix for the problem.

**Reply by fforootd** [reply]
I think thats a valid workaround since it triggers all the builds in x86 instead of arm. (even though the build itself should work with arm)

**Reply by usaaama2010** [reply]
executing goreleaser command followed by `GOARCH=amd64 GOOS=linux` also works fine on Apple Silicon
